Source/Plugins/Core/com.equella.core/js/__tests__/tsrc/search/components/SearchResult.test.tsx

```diff
@@ -172,19 +191,97 @@ describe("<SearchResult/>", () => {
     );
   });
 
-  it.each<[string]>([
-    [languageStrings.searchpage.selectResource.summaryPage],
-    [languageStrings.searchpage.selectResource.allAttachments],
-    [languageStrings.searchpage.selectResource.attachment],
-  ])(
-    // todo: pass event handler in and check if it has been called.
-    "should display buttons for %s in Selection Session",
-    async (buttonLabel: string) => {
-      updateMockRenderData();
-      const { queryByLabelText } = await renderSearchResult(
-        mockData.attachSearchObj
-      );
-      expect(queryByLabelText(buttonLabel)).toBeInTheDocument();
-    }
-  );
+  describe("In Selection Session", () => {
+    const mockGlobalCourseList = jest.spyOn(
+      LegacySelectionSessionModule,
+      "getGlobalCourseList"
+    );
+    mockGlobalCourseList.mockReturnValue({ updateCourseList: jest.fn() });
+
+    const mockSelectResourceForCourseList = jest.spyOn(
+      LegacySelectionSessionModule,
+      "selectResourceForCourseList"
+    );
+    mockSelectResourceForCourseList.mockResolvedValue();
+
+    const mockSelectResourceForNonCourseList = jest.spyOn(
+      LegacySelectionSessionModule,
+      "selectResourceForNonCourseList"
+    );
+    mockSelectResourceForNonCourseList.mockResolvedValue();
+
+    const STRUCTURED = "structured";
+    const SELECT_OR_ADD = "selectOrAdd";
+    const selectForCourseFunc = selectResourceForCourseList;
+    const selectForNonCourseFunc = selectResourceForNonCourseList;
+
+    const makeSelection = (
+      selectorLabel: string,
+      queryByLabelText: (text: string) => HTMLElement | null,
+      getByLabelText: (text: string) => HTMLElement
+    ) => {
+      // First, make sure the selector control is active
+      expect(queryByLabelText(selectorLabel)).toBeInTheDocument();
+      // And then make a selection by clicking it
+      fireEvent.click(getByLabelText(selectorLabel));
+    };
```

Comment:
This could be simplified with:

```suggestion
    const makeSelection = (
      findSelector: (text: string) => HTMLElement | null,
    ) => {
      const selectorControl = findSelector();
      // First, make sure the selector control is active
      expect(selectorControl).toBeInTheDocument();
      // And then make a selection by clicking it
      fireEvent.click(selectorControl);
    };
```

And that way when you use it you'd only have to:

```typescript
makeSelection(() => queryByLabelText(resourceType));
```

Reply:
But I guess `fireEvent.click` will complain because `selectorControl` could be null.

Reply:
Yeah, here is the error: 

Argument of type 'HTMLElement | null' is not assignable to parameter of type 'Element | Node | Window | Document'.
      Type 'null' is not assignable to type 'Element | Node | Window | Document'.

What about we use a if statement and throw an error if `queryByLabelText` returns null ? We have used this approach for `querySelector` in a few places.

Reply:
Yup, I'd say that'd be the plan. But just wrap the fireEvent in the `if`.

Actually, this is probably a valid case of using the not null (`!`) postfix operator - as we definitely know (due to validation with `expect`) that this is non-null. :thinking: